src/typeahead/typeahead.js

```diff
@@ -399,13 +399,29 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
     };
   }])
 
-  .filter('typeaheadHighlight', function() {
+  .filter('typeaheadHighlight', ['$sce', '$injector', '$log', function($sce, $injector, $log) {
+
+    var $sanitize;
+    try {
+      $sanitize = $injector.get('$sanitize'); //tries to inject the sanitize service
+    }
+    catch(e) {
+      $log.warn('Unsafe use of typeahead please use ngSanitize'); // Warn the user about the danger
+    }
 
     function escapeRegexp(queryToEscape) {
+      // Regex: capture the whole query string and replace it with the string that will be used to match 
+      // the results, for example if the capture is "a" the result will be \a
       return queryToEscape.replace(/([.?*+^$[\]\\(){}|-])/g, '\\$1');
     }
 
     return function(matchItem, query) {
-      return query ? ('' + matchItem).replace(new RegExp(escapeRegexp(query), 'gi'), '<strong>$&</strong>') : matchItem;
+      matchItem = query? ('' + matchItem).replace(new RegExp(escapeRegexp(query), 'gi'), '<strong>$&</strong>') : matchItem; // Replaces the capture string with a the same string inside of a "strong" tag
```

Comment:
Sorry, on further thought, I realized your initial logic to check for html tags should be used here like:
```js
if (containsHtml(matchItem) && !$sanitize) {
    $log.warn('Unsafe use of typeahead please use ngSanitize');
}
```

This way, the user will only be warned when using typeahead with labels that contain HTML.